contracts/oraclize/OraclizeWrapper.sol

```diff
@@ -23,18 +24,18 @@ contract OraclizeWrapper is OraclizeI, UsingOracleI {
         UsingOraclizeI(callbackAddress).__callback(_id, _value);
     }
 
-    function query(uint _timestamp, string _datasource, string _arg) external payable returns (bytes32 _id) {
+    function query(uint _timestamp, string _datasource, string _arg) external payable onlyRole(AUTHORIZED_USER) returns (bytes32 _id) {
```

Comment:
The "authorized user" might actually be a contract so the term "user" can be a little misleading. How about "AUTHORIZED_ROLE"?